src/checkstatus/SSMAutomationExecutionStatusWaiter.py

```diff
@@ -0,0 +1,31 @@
+import botocore
+import constants
+import waiter_acceptor_config
+
+def check_automation_execution_status(automation_execution_id, max_attempts, ssm):
+    acceptor_config = waiter_acceptor_config.SSM_AUTOMATION_EXECUTION_COMPLETED
+    model = botocore.waiter.WaiterModel({
+        "version": 2,
+        "waiters": {
+            "SSMAutomationExecutionStatus": {
+                "delay": constants.DELAY_INTERVAL,
+                "operation": "DescribeAutomationExecutions",
+                "maxAttempts": max_attempts,
+                "acceptors": acceptor_config
+            }
+        }
+    })
+
+    automation_execution_filter = [
```

Comment:
Can this filter be optimised to return specific fields of metadata needed ?

Reply:
no thats not supported - it just defines which method to be invoked and the waiter_acceptor_config validates against the required field in the response

Reply:
ok , strange that a filter is required to getById

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ssm/client/describe_automation_executions.html - its the method signature

Reply:
You can see  https://boto3.amazonaws.com/v1/documentation/api/1.26.93/reference/services/ssm/client/get_automation_execution.html which gives directly by id

Reply:
ya i had tried this too - get_automation_execution did not work with the waiter so i changed it to describe considering the cost of both was the same and that since Id is being provided, the response size is known beforeahnd

Reply:
ok